### DISCUSSION TITLE
Add support for `bun install`

### DISCUSSION THREAD

**Discussion by MarkLyck** [vercel/community#2021] [Feature Question] [answered]
I've recently switched my projects from using `npm`, `yarn` or `pnpm` to `bun`

https://bun.sh/

bun is significantly faster at installing node_modules than the other package managers. But it is not yet supported by Vercel.

Please add support for `bun install`

<img width="802" alt="Screenshot 2023-04-09 at 19 13 00" src="https://user-images.githubusercontent.com/6841110/230786660-200aec27-13ce-4d86-a647-1551b674a2a2.png">

**Comment by ghost** [comment]
Looking for the same here

**Comment by loicnestler** [comment]
CC'ing @oven-sh here, maybe they know on how to include the bun runtime in your Vercel bundle. 👀

**Comment by Jarred-Sumner** [comment]
@leerob let me know if there’s anything we can do to help

**Reply by styfle** [reply]
@Jarred-Sumner What is the plan for lock file versioning? 

It looks like a version is stored in [bun.lockb](
https://github.com/oven-sh/bun/blob/51d3d4382281f789f8175079ed426a63529eb3e7/src/install/lockfile.zig#L4118-L4119) but its not clear if this will follow the version of bun? 

For example, does `bun-lockfile-format-v0` represent Bun 1.0? Will Bun 2.0 bump the lock file to `bun-lockfile-format-v1`?

**Reply by gtrabanco** [reply]
You can view the content by executing:

```
bun ./bun.lockb
```

**Comment by hyuzea** [comment]
do `curl -fsSL https://bun.sh/install | bash`  before `bun i`,  maybe

**Reply by alvarlagerlof** [reply]
Can you do that inside Vercel?

**Reply by alvarlagerlof** [reply]
I suspected this would not be allowed, and it is not. 

![image](https://github.com/vercel/community/assets/14835120/6900eb94-90ca-40c3-ae00-3bb2d9437a7f)

**Reply by hyuzea** [reply]
@alvarlagerlof, for installs with curl, you have to run bun as `~/.bun/bin/bun`.
BTW `npm  i -g bun` can be used too and you can just run `bun` without prefix

![image](https://github.com/vercel/community/assets/3823496/6acce468-627f-4271-85ad-063f0295fcc2)

**Reply by gabrielmaldi** [reply]
This is working for me:

`vercel.json`
```json
{
  "$schema": "https://openapi.vercel.sh/vercel.json",
  "installCommand": "curl -fsSL https://bun.sh/install | bash && ~/.bun/bin/bun install --frozen-lockfile",
  "buildCommand": "~/.bun/bin/bun run build"
}
```

However, I think something has to be done for the build cache to work with bun. Even though I get this output:
```sh
Uploading build cache [177.05 MB]...
Build cache uploaded: 2.985s
```
On the next build I get:
```sh
Skipping build cache since Package Manager changed from "pnpm" to "yarn"
```
So even though it's generated, the build cache is not being used.

**Comment by RickRyan26** [comment]
Bun 1.0 just dropped!

@StephDietz any chance you're bringing a SvelteKit Vercel Bun adapter to life??

https://bun.sh/blog/bun-v1.0

https://bun.sh/guides/ecosystem/sveltekit

**Reply by silasabbott** [reply]
Bun support for SvelteKit on Vercel would probably be best added to @sveltejs/adapter-vercel once Vercel supports Bun as a runtime

**Comment by RatebSeirawan** [comment]
I only have good things to say about Bun so far!
I'm using it in development with nextjs but I'd love to have it running with my deployment on vercel

**Comment by andres-carvajal** [comment]
https://maximevermeeren.medium.com/how-to-use-bun-on-vercel-7512383153d7

**Reply by RickRyan26** [reply]
A few hours ago leerob said it can't be done yet. Hmmm

**Reply by mquandalle** [reply]
Well he's hinting at upcoming support on Vervel https://twitter.com/leeerob/status/1701031958720254159

**Reply by boredland** [reply]
Not sure what @andres-carvajal proposes supports caching yet (relatively sure it doesn't), but it certainly could be a workaround until real support is there.

**Reply by boredland** [reply]
Hm, dependency cache seems to work just fine already? A bit confused why tho.

**Comment by miguderp** [comment]
Needless to say, I'm sure a ton of developers are looking forward to the implementation of Bun on Vercel, me included. I'd say that, for once, the hype is 100% healthy hype

**Comment by bujoralexandru** [comment]
Indeed, Bun has really shown great promise and, looking at the benchmarks (I know, benchmarks can be shallow) the team delivered.

**Comment by LZL0** [comment]
Let's go!

**Comment by leerob** [comment]
We definitely want to support Bun on Vercel. We'll update this discussion as we know more 😄

**Reply by DelightfulMoose6158** [reply]
👀

**Reply by LZL0** [reply]
Let's go!! 🔥

**Reply by franrull** [reply]
Watching! 🍿 👀

**Reply by troymcginnis** [reply]
👀

**Reply by ngdluan** [reply]
cool!

**Reply by mzbali** [reply]
waiting!

**Reply by itsmikesharescode** [reply]
any update??

**Reply by TooTallNate** [reply]
@itsmikesharescode It's been supported for a while https://github.com/orgs/vercel/discussions/2021#discussioncomment-6973207

**Comment by santosant** [comment]
W

**Comment by styfle** [comment]
If you are interested in using Bun as a package manager via Corepack, upvote the linked issue:

- https://github.com/nodejs/corepack/issues/295

**Comment by antonyayansi** [comment]
Bun 🧄

**Comment by jesusmartinoza** [comment]
Raid from @midudev 😎

**Comment by Spikeysanju** [comment]
### 🔊 Update
Vercel now supports Bun installation with no setup needed. This change affects the build phase but doesn't affect the runtime. So, Serverless Functions won't use the Bun runtime for now.

official tweet: https://twitter.com/vercel/status/1701378595623415934?s=20

**Comment by leerob** [comment, marked as answer]
Good news, Vercel now supports Bun install with zero configuration!

https://vercel.com/changelog/bun-install-is-now-supported-with-zero-configuration

**Reply by arvinxx** [reply]
Is there anyway to use `bun install` without lockfile ?  We want to install the latest depenencies  always with bun.

**Reply by trek** [reply]
@arvinxx There is! If you define a [custom install command](https://vercel.com/docs/deployments/configure-a-build#install-command) it should work. The Bun binary is currently located at `/bun1/bun`

**Reply by arvinxx** [reply]
> @arvinxx There is! If you define a [custom install command](https://vercel.com/docs/deployments/configure-a-build#install-command) it should work. The Bun binary is currently located at `/bun1/bun`

Maybe the Bun binary should add to Path？ Or add usage in document.

**Reply by JosephTLyons** [reply]
Does it actually speed up dependency installations in vercel?  Does the bun cache work over there?  I couldn't tell the difference between using `npm nstall` and using `bun install`, in vercel, even on subsequent runs. Locally, I would see massive speedups on subsequent `bun install` runs, as the cache was being used, but I didn't see a diffence in the actual deployment of my site.

**Reply by arvinxx** [reply]
@JosephTLyons yes, our install is from 60s to 6s

**Reply by JosephTLyons** [reply]
Damn, that's a huge win! Ours is still painfully slow, for some reason.

**Reply by rubas** [reply]
I would be amazing, if `bunx` would also be available as we use it in `postinstall`.

```
/usr/bin/bash: bunx: command not found
--
error: script "postinstall" exited with code 127
```

**Comment by rubenmarcus** [comment]
tried to move a legacy next.js 12 from npm to bun, but lot of packages like firebase, also Dynamic Imports and third party library components didnt work without ts-ignore it complains it is not a valid JSX component...

no success on running bun ... also firebase package complained when running build in a client-side firebase service.

**Reply by gtrabanco** [reply]
Vercel only supports `bun install`.

Anyway if you have errors while running any npm package just report it on Bun repository. They are trying hard to have full or almost full support for all stuff that Node can run.

**Reply by KBeDevel** [reply]
Adding to @gtrabanco, you may also want to check the [Node API compatibility state](https://bun.sh/docs/runtime/nodejs-apis). Sadly, not all APIs are implemented yet and, as long as I know, Bun is based on [JSC](https://developer.apple.com/documentation/javascriptcore) and Firebase features may not to be fully supported.

**Comment by BrunoSette** [comment]
```
[00:02:35.003] Running build in Washington, D.C., USA (East) – iad1 (Hive)
[00:02:35.394] Cloning github.com/xxxxxxxxxxx/xxxxxxxx (Branch: main, Commit: 120f5f0)
[00:02:35.404] Skipping build cache, deployment was triggered without cache.
[00:02:37.410] Cloning completed: 2.015s
[00:02:37.901] Running "vercel build"
[00:02:40.044] Vercel CLI 32.2.5
[00:02:41.089] Running "install" command: `bun install`...
[00:02:41.098] sh: bun: command not found
[00:02:41.106] Error: Command "bun install" exited with 127
```

I'm getting the error: 
```
sh: bun: command not found
Error: Command "bun install" exited with 127
```

**Reply by cipriancaba** [reply]
We are just trying to migrate to `bun` but it doesn't work, we get the same bun command not found error even though we cleared the cache. It bun install command gets picked automatically.

@leerob 

```


Skipping build cache, deployment was triggered without cache.
--

16:18:15.577 | Running build in Cleveland, USA (East) – cle1
16:18:16.859 | Cloning completed: 1.282s
16:18:16.859 | Running "bash vercel.sh"
16:18:16.878 | VERCEL_GIT_COMMIT_REF: dev
16:18:16.878 | ✅ - Build can proceed
16:18:17.200 | Running "vercel build"
16:18:17.649 | Vercel CLI 32.2.5
16:18:17.771 | > Detected Turbo. Adjusting default settings...
16:18:17.916 | Running "install" command: `bun install`...
16:18:17.922 | sh: bun: command not found
16:18:17.927 | Error: Command "bun install" exited with 127


```

**Reply by chhuang** [reply]
Did you update `packageManager: bun@1.0.3` in the root `package.json`? Try removing the local `bun.lockb`, run `bun i` and redeploy.

**Reply by cipriancaba** [reply]
Thanks for the suggestion @chhuang Still same error even with the changes you suggested. Tried clearing the entire vercel cache and it didn't help

**Comment by oagix** [comment]
For some reason `bun install` always uses version 1.0.6 instead of the latest one or the one specified in package.json (i.e. 1.0.14) on Vercel. Any insights on how to get around that?

**Reply by notxorand** [reply]
Experiencing the same, for now I'm just using Yarn. I love Bun but being relatively new seems pretty problematic.

**Comment by applemate** [comment]
so, we gonna get bun run time, soon ?

**Comment by blaisewf** [comment]
Please, bun as a runtime soon 🙏🏻

**Comment by itsmikesharescode** [comment]
bun runtime!!!

**Comment by mxmzb** [comment]
Bun on Vercel is currently stuck @v1.1.17 for me. Can we upgrade please, because [.npmrc support is available from 1.1.18](https://github.com/oven-sh/bun/issues/643#issuecomment-2198393603) on.

[a comment by an automated account is not shown]
